src/redux/slices/api.js

```diff
@@ -0,0 +1,538 @@
+import { createApi } from '@reduxjs/toolkit/query/react';
+
+import getDifferentiatorSuffix from '../../lib/get-differentiator-suffix.js';
+import pruneInstance from '../../lib/prune-instance.js';
+import { deactivateNotification } from '../action-handlers/notification.js';
+import {
+	errorActivated,
+	notificationActivated,
+	redirectActivated
+} from '../actions/index.js';
+import {
+	ACTIONS,
+	MODELS,
+	MODEL_TO_DISPLAY_NAME_MAP,
+	MODEL_TO_PASCAL_CASE_MAP,
+	MODEL_TO_ROUTE_MAP,
+	NOTIFICATION_STATUSES,
+	PLURALISED_MODELS,
+	PLURALISED_MODEL_TO_ROUTE_MAP
+} from '../../utils/constants.js';
+
+const API_BASE_URL = 'http://localhost:3000';
+
+const baseQuery = async ({ apiRoute, fetchSettings = {} }, { dispatch }) => {
+
+	try {
+
+		const apiUrl = `${API_BASE_URL}${apiRoute}`;
+
+		const response = await fetch(apiUrl, { ...fetchSettings, mode: 'cors' });
+
+		if (response.status !== 200) throw new Error(response.statusText);
+
+		const responseJson = await response.json();
+
+		return { data: responseJson };
+
+	} catch (error) {
+
+		dispatch(errorActivated({ message: error.message }));
+
+		dispatch(deactivateNotification());
+
+		return { error: error.toString() };
```

Comment:
If the following is tried then the below console errors will appear:
```js
return { error };
```

> “A non-serializable value was detected in an action, in the path: `payload`. Value: Error: Not Found”

> “A non-serializable value was detected in the state, in the path: `api.queries.getSeason("foo").error`. Value: Error: Not Found”

ChatGPT's response:
> The Error object is often considered a non-serializable value, especially in the context of the state or action payload. When you pass an Error object (such as new Error('Not Found')), it’s treated as non-serializable, leading to the warning you're seeing

Alternate options that would also make the `error` value serializeable:

```js
return { error: { message: error.message, stack: error.stack } };
```
```js
return { error: error.message };
```